main.c

```diff
@@ -1,11 +1,73 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 #include "mzcc.h"
 
+char *outfile = NULL, *infile = NULL;
+extern FILE *outputfp;
+bool dump_ast;
+
+static void usage()
+{
+    fprintf(stdout,
+            "Usage : mzcc <input file>\n"
+            "\n"
+
+            "-o filename     Output to the specified file\n"
+            "-dump-ast       Print Abstract Syntax Tree\n");
+    exit(1);
+}
+static void parseopt(int argc, char **argv)
+{
+    if (argc == 1) {
+        usage();
+    } else {
+        int opt;
+        while ((opt = getopt(argc, argv, "o:d:")) != -1) {
```

Comment:
The `getopt()`  function is provided by POSIX.1-2001, POSIX.1-2008, and POSIX.2 with environment variable `POSIXLY_CORRECT`. I wish to avoid the use of `getopt` for such simple cases. You might check how AMaCC manipulates the arguments: https://github.com/jserv/amacc